consensus/src/quorum_store/batch_aggregator.rs

```diff
@@ -117,6 +117,9 @@ impl BatchAggregator {
 
     fn is_outdated_fragment(&self, batch_id: BatchId, fragment_id: usize) -> bool {
         if let Some(self_batch_id) = self.batch_id {
+            if self_batch_id.nonce != batch_id.nonce {
```

Comment:
Can you add a comment here to explain? like if nonce is different, then it is a new batch (after recovery) so should not be outdated.